ecl/hql/hqlgram.y

```diff
@@ -7976,27 +7976,29 @@ simpleDataSet
                             $$.setExpr(createDataset(no_choosesets, $3.getExpr(), $5.getExpr()));
                             $$.setPosition($1);
                         }
-    | DEDUP '(' startTopLeftRightSeqFilter optDedupFlags ')' endTopLeftRightFilter endSelectorSequence
+    | DEDUP '(' startSortOrder startTopLeftRightSeqFilter optDedupFlags ')' endTopLeftRightFilter endSelectorSequence endSortOrder
```

Comment:
I'm not sure it makes sense either...
May have said as much offline when talking to Shamser already.

Generally though, did you envisage BEST() always representing a field set?

I pondered whether e.g.
```
DEDUP(ds, surname, BEST(age=>13 && age<=18));
```
Would mean dedup by surname, keeping the 1st that met compare expression age>=13 && age<=18, or last who did if RIGHT.